src/app/auto-update-handler.ts

```diff
@@ -1,67 +1,119 @@
 
-    this.updateUtilArgs = [];
+      this.autoUpdater.on('update-downloaded', () => {
+        this.isUpdateAvailable = true;
+        const mainWebContents = windowHandler.mainWebContents;
+        // Display client banner
+        if (mainWebContents && !mainWebContents.isDestroyed()) {
+          mainWebContents.send('display-client-banner', {
+            reason: 'autoUpdate',
+            action: 'update-available',
+          });
+        }
+      });
+    }
   }
 
   /**
-   * Launch the auto update helper
+   * Installs the latest update quits and relaunches application
+   */
+  public updateAndRestart = async (): Promise<void> => {
+    if (!this.isUpdateAvailable) {
+      return;
+    }
+    // Handle update and restart for macOS
+    if (isMac) {
+      windowHandler.setIsAutoUpdating(true);
+    }
+    setImmediate(() => {
+      if (this.autoUpdater) {
+        this.autoUpdater.quitAndInstall();
+      }
+    });
+  };
+
+  /**
+   * Checks for the latest updates
+   * @return void
+   */
+  public checkUpdates = async (): Promise<void> => {
+    logger.info('auto-update-handler: Checking for updates');
+    if (this.autoUpdater) {
+      const updateCheckResult = await this.autoUpdater.checkForUpdatesAndNotify();
+      logger.info('auto-update-handler: ', updateCheckResult);
+    }
+    logger.info('auto-update-handler: After checking auto update');
+  };
+
+  /**
+   * Constructs the SDA auto update end point url
+   *
+   * @return string
+   * @example https://corporate.symphony.com/macos/general
    */
-  public async update(filename: string) {
-    logger.info(`auto-update-handler: Starting auto update!`);
-    this.filename = filename;
-    if (isWindowsOS) {
-      this.updateUtilArgs = [
-        this.filename,
-        app.getPath('exe'),
-        this.logFilePath,
-      ];
+  public getUpdateUrl = (): string => {
+    const { url: userConfigURL } = config.getUserConfigFields(['url']);
+    const { url: globalConfigURL } = config.getGlobalConfigFields(['url']);
+    const { autoUpdateUrl, autoUpdateChannel } = config.getConfigFields([
+      'autoUpdateChannel',
+      'autoUpdateUrl',
+    ]);
+
+    if (autoUpdateUrl) {
       logger.info(
-        `auto-update-handler: Running update helper${this.updateUtil} with args ${this.updateUtilArgs}!`,
+        `auto-update-handler: autoUpdateUrl exists so, using it`,
+        autoUpdateUrl,
       );
-
-      try {
-        spawn(this.updateUtil, this.updateUtilArgs);
-      } catch (error) {
-        logger.error(
-          `auto-update-handler: auto update failed. Error: ${error}!`,
-        );
-      }
+      return autoUpdateUrl;
     }
-  }
+
+    const url = userConfigURL ? userConfigURL : globalConfigURL;
+    const os = isMac ? MACOS : WINDOWS;
+
+    const { subdomain, domain, tld } = whitelistHandler.parseDomain(url);
+    logger.info(
+      `auto-update-handler: using generic pod url`,
+      `https://${subdomain}.${domain}.${tld}/${os}/${autoUpdateChannel}`,
```

Comment:
It might be worth having a dedicated variable for the URL and reusing it as a return value.